File: 백준/Bronze/9506. 약수들의 합/약수들의 합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Collections;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
				
		while(true) {
			StringBuilder sb = new StringBuilder();
			int n = Integer.parseInt(br.readLine());
			
			if(n == -1)
				return;
			
			ArrayList<Integer> al = new ArrayList<Integer>();
			
			for(int i = 1; i <= Math.sqrt(n); i++) {
				if(n % i == 0) {
					al.add(i);
					al.add(n / i);
				}
			}
			
			Collections.sort(al);

			int sum = 0;
			
			sb.append(n + " = ");
			for(int i = 0; i < al.size() - 1; i++) {
				int v = al.get(i);
				sum += v;
				
				sb.append(v + " + ");
			}
			
			if(sum == n) {
				sb.delete(sb.length()-3, sb.length()-1);
				System.out.println(sb);
			}
			else
				System.out.println(n + " is NOT perfect.");
		}	
	}
}

File: 백준/Bronze/9506. 약수들의 합/약수들의 합.js
const fs = require('fs');
let input = fs.readFileSync('./dev/stdin').toString().trim().split("\n").map(e => e.trim());

for (let i = 0; i < input.length - 1; i++) {
    const n = parseInt(input[i]);
    const set = new Set();
    for (let j = 1; j <= Math.sqrt(parseInt(input[i])); j++) {
        if (n % j == 0) {
            set.add(j);
            set.add(n / j);
        }
    }

    const sort = [...set].sort((a, b) => a - b).slice(0, -1);
    const sum = sort.reduce((acc, cur) => acc + cur, 0);
    if (n === sum) {
        console.log(`${n} =`, sort.join(' + '));
    } else {
        console.log(`${n} is NOT perfect.`);
    }
}